testbed/infrastructure.ts

```diff
@@ -0,0 +1,47 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as eks from '@aws-cdk/aws-eks';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as cdk from '@aws-cdk/core'
+import { StackProps } from '@aws-cdk/core';
+
+export class Infrastructure extends cdk.Construct {
+    cluster: eks.Cluster;
+    constructor(scope: cdk.Construct, id: string, props: StackProps) {
+        super(scope, id);
+
+        // A VPC, including NAT GWs, IGWs, where we will run our cluster
+        const vpc = new ec2.Vpc(this, 'VPC', {});
+
+        // The EKS cluster
+        this.cluster = new eks.Cluster(this, 'Cluster', {
+            vpc: vpc,
+            role: new iam.Role(this, 'ClusterRole', {
+                assumedBy: new iam.ServicePrincipal('eks.amazonaws.com'),
+                managedPolicies: [
+                    iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSClusterPolicy'),
+                    iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSVPCResourceController')
+                ]
+            }),
+            version: eks.KubernetesVersion.V1_19,
+            defaultCapacity: 0
+        });
+
+        // for time being we start with these defaults; To-do use karpenter to manage this
+        this.cluster.addNodegroupCapacity('WorkerNodeGroup', {
```

Comment:
Any reason to not chain directly onto line 27?

Reply:
If you are asking if I can chain our customized workernode group stuff into cluster creation above, it's not possible AFSIU, only possibility is to choose defaults which I don't think I want at this point.